client/src/rpb/granular_view.js

```diff
@@ -153,112 +68,96 @@ class GranularView extends React.Component {
-        .map(({ nick, type, fully_qualified_name }, idx) => [
+        .map(({ nick, type, fully_qualified_name, initial_visible }, idx) => [
           nick,
           {
             index: idx + 1,
             header: fully_qualified_name,
             is_searchable: !is_matched_undefined(non_dept_key_cols, nick),
             is_summable: !is_matched_undefined(data_columns, nick),
             formatter: type,
+            initial_visible:
+              !is_matched_undefined(non_dept_key_cols, nick) || initial_visible,
           },
         ])
         .fromPairs()
         .value(),
     };
-    const table_data = _.map(shown_rows, (row) => ({
+    const table_data = _.map(flat_data, (row) => ({
       dept: Dept.lookup(row.dept).name,
       ..._.chain(cols)
         .map(({ nick }) => [nick, row[nick]])
         .fromPairs()
         .value(),
     }));
+    const group_filter_options = _.map(
+      filters_by_dimension,
+      (filter_by_dim) => ({
+        ...filter_by_dim,
+        children: _.sortBy(filter_by_dim.children, "display"),
+      })
+    );
 
     return (
-      <div>
-        <DisplayTable data={table_data} column_configs={column_configs} />
-
-        {!excel_mode && pages.length > 1 && (
-          <div className="pagination-container">
-            {window.is_a11y_mode && (
-              <p>
-                <TextMaker
-                  text_key="pagination_a11y"
-                  args={{ current: page_num, total: pages.length }}
-                />
-              </p>
-            )}
-            <ul className="pagination">
-              {_.map(pages, (data, ix) => (
-                <li
-                  key={ix}
-                  className={classNames(ix === page_num && "active")}
-                >
-                  <span
-                    tabIndex={0}
-                    style={
-                      ix === page_num
-                        ? {
-                            color:
-                              window.infobase_color_constants.textLightColor,
-                          }
-                        : null
-                    }
-                    disabled={page_num === ix}
-                    role="button"
-                    onClick={
-                      ix === page_num
-                        ? null
-                        : () => {
-                            on_set_page(ix);
-                            this.refs.table.focus();
-                          }
-                    }
-                    onKeyDown={
-                      ix === page_num
-                        ? null
-                        : (e) => {
-                            if (e.keyCode === 13 || e.keyCode === 32) {
-                              on_set_page(ix);
-                              this.refs.table.focus();
+      <DisplayTable
+        data={table_data}
+        column_configs={column_configs}
+        util_components={{
+          rpb_group_data: (
+            <DropdownMenu
```

Comment:
Big JSX pyramid below here. Could possibly define inner chunks as separate components to break up and flatten this a bit?